form/3d/basiljs/cube/cube.jsx

```diff
@@ -1 +1,5 @@
  var xminus2D = convertToScreen(xminus3D);
  var yplus2D = convertToScreen(yplus3D);
  var yminus2D = convertToScreen(yminus3D);
  var zplus2D = convertToScreen(zplus3D);
  var zminus2D = convertToScreen(zminus3D);
  var line1 = b.line(xplus2D[0].x, xplus2D[0].y, xplus2D[1].x, xplus2D[1].y);
  var txt1 = b.text('+x', xplus2D[1].x, xplus2D[1].y, 20, 20);
  var line2 = b.line(xminus2D[0].x, xminus2D[0].y, xminus2D[1].x, xminus2D[1].y);
  var txt2 = b.text('-x', xminus2D[1].x, xminus2D[1].y, 20, 20);
  var line3 = b.line(yplus2D[0].x, yplus2D[0].y, yplus2D[1].x, yplus2D[1].y);
  var txt3 = b.text('+y', yplus2D[1].x, yplus2D[1].y, 20, 20);
  var line4 = b.line(yminus2D[0].x, yminus2D[0].y, yminus2D[1].x, yminus2D[1].y);
  var txt4 = b.text('-y', yminus2D[1].x, yminus2D[1].y, 20, 20);
  var line5 = b.line(zplus2D[0].x, zplus2D[0].y, zplus2D[1].x, zplus2D[1].y);
  var txt5 = b.text('+z', zplus2D[1].x, zplus2D[1].y, 20, 20);
  var line6 = b.line(zminus2D[0].x, zminus2D[0].y, zminus2D[1].x, zminus2D[1].y);
  var txt6 = b.text('-z', zminus2D[1].x, zminus2D[1].y, 20, 20);
  return [line1, line2, line3, line4, line5, line6, txt1, txt2, txt3, txt4, txt5, txt6];
}

function draw() {
  b.clear(b.doc()); // clear the current document
  b.units(b.MM); // we want to print. use MM instead of default pixels
  var doc = b.doc(); // a reference to the current document
  // set some preferneces of the document for better handling
  doc.documentPreferences.properties = {
    pageWidth: pw,
    pageHeight: ph
  }; // set the page size
  doc.viewPreferences.rulerOrigin = RulerOrigin.SPREAD_ORIGIN; // upper left corner
  // ----------
  // main code goes here
  // define some 3D points
  b.noFill();
  // var makeACube = function (x,y,z,w,h,d){
  //b.pushMatrix(); // push the matrix
  // b.translate(x, y); // to the center of the screen
  // var c = cube(x,y,z,w,h,d);
  // draw the top plane. Could also be a polygon
  //b.popMatrix();// reset the matrix
  // return c;
  // }
  //~   b.pushMatrix(); // push the matrix
  //~   b.translate(b.width/2, b.height/2); // to the center of the screen
  //~   var planes = cube(50,100,100);
  // draw the top plane. Could also be a polygon
  //~   b.popMatrix();// reset the matrix
  doc.pages[0].groups.add(cube(b.width / 2, b.height / 2, -200, 50, 50, 50));
  doc.pages[0].groups.add(cube(b.width / 2, -(b.height / 2), -200, 50, 50, 50));
  //doc.pages[0].groups.add(makeACube (b.width/4,b.height/2, 50,50,50));
  // doc.pages[0].groups.add(makeACube ((b.width/4)*3,b.height/2, 50,50,100));
  //b.pushMatrix();
  //b.translate(b.width/2,b.height/2);
  b.fill(0);
  doc.pages[0].groups.add(axis());
  //b.popMatrix();
  // ----------
  var fname = File($.fileName).parent.fsName + '/' + ($.fileName.split('/')[$.fileName.split('/').length - 1]).split('.')[0] + '.indd';
  // b.println(fname);
  doc.save(fname, false, 'basil', true);
  b.savePNG('out.png');
}
b.go();
\ No newline at end of file
+<<<<<<< HEAD
```

Comment:
Something went totally wrong with this file. You shouldn't even have touched this nor committed it.